runner/operator/access/v1_0_0/merge_fastqs/__init__.py

```diff
@@ -0,0 +1,111 @@
+import os
+from collections import defaultdict
+from itertools import groupby
+from runner.operator.operator import Operator
+from runner.serializers import APIRunCreateSerializer
+from file_system.repository.file_repository import FileRepository
+
+import json
+from jinja2 import Template
+
+WORKDIR = os.path.dirname(os.path.abspath(__file__))
+
+
+METADATA_OUTPUT_FIELDS = [
+    'barcodeId',
+    'sampleName',
+    'investigatorSampleId',
+    'patientId',
+    'tumorOrNormal',
+    'sampleOrigin',
+    'dnaInputNg',
+    'captureInputNg',
+    'baitSet',
+    'sex',
+    'barcodeIndex',
+    'libraryVolume',
+    'captureName',
+    'libraryConcentrationNgul'
+]
+def construct_inputs(samples):
+    with open(os.path.join(WORKDIR, 'input_template.json.jinja2')) as file:
+        template = Template(file.read())
+
+
+    samples = list(group_by_sample_id(samples).values())
+
+    inputs = list()
+    for sample_files in samples:
+        fastqs = group_by_fastq(sample_files)
+
+        fastq1s = [{
+            "class": "File",
+            "location": "juno://" + fastq["path"]
+        } for fastq in fastqs["R1"]]
+
+        fastq2s = [{
+            "class": "File",
+            "location": "juno://" + fastq["path"]
+        } for fastq in fastqs["R2"]]
+
+        input_file = template.render(
+            fastq1_files=json.dumps(fastq1s),
+            fastq2_files=json.dumps(fastq2s),
+        )
+
+        inputs.append((json.loads(input_file), sample_files[0]["metadata"]))
```

Comment:
The problem with using `sample_files[0]` here is that there can be different sample ID info for the two pairs of fastqs (because they came from different sequencing runs)

For example these fields might be different between [0] and [1]:
```
    'dnaInputNg',
    'captureInputNg',
    'libraryVolume',
    'libraryConcentrationNgul'
```

In the `lims_client` I made some logic to take the average of these values:

https://github.mskcc.org/MSK-ACCESS/lims_client/blob/master/lims_client/models.py#L128-L151

But it should be ok for now unless Ronak has any concerns

Reply:
Ah I forgot about that. I'll average them out.